engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/BusinessRuleTaskTest.java

```diff
@@ -334,4 +336,86 @@ public void shouldWriteDecisionEvaluationEvent() {
                       });
             });
   }
+
+  @Test
+  public void shouldWriteDecisionEvaluationEventIfEvaluationFailed() {
+    // given
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlClasspathResource(DMN_RESOURCE)
+            .withXmlResource(
+                processWithBusinessRuleTask(
+                    t ->
+                        t.zeebeCalledDecisionId("force_user").zeebeResultVariable(RESULT_VARIABLE)))
+            .deploy();
+
+    final var calledDecision =
+        deployment.getValue().getDecisionsMetadata().stream()
+            .filter(decision -> decision.getDecisionId().equals("force_user"))
+            .findFirst()
+            .orElseThrow();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    final var businessRuleTaskActivating =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATING)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementType(BpmnElementType.BUSINESS_RULE_TASK)
+            .getFirst();
+
+    // then
+    final var decisionEvaluationRecord =
+        RecordingExporter.decisionEvaluationRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    assertThat(decisionEvaluationRecord)
+        .hasRecordType(RecordType.EVENT)
+        .hasValueType(ValueType.DECISION_EVALUATION)
+        .hasIntent(DecisionEvaluationIntent.FAILED)
+        .hasSourceRecordPosition(businessRuleTaskActivating.getSourceRecordPosition());
+    assertThat(decisionEvaluationRecord.getKey())
+        .describedAs("Expect that the decision evaluation event has a key")
+        .isPositive();
+
+    final var decisionEvaluationValue = decisionEvaluationRecord.getValue();
+    assertThat(decisionEvaluationValue)
+        .hasDecisionKey(calledDecision.getDecisionKey())
+        .hasDecisionId(calledDecision.getDecisionId())
+        .hasDecisionName(calledDecision.getDecisionName())
+        .hasDecisionVersion(calledDecision.getVersion())
+        .hasDecisionRequirementsKey(calledDecision.getDecisionRequirementsKey())
+        .hasDecisionRequirementsId(calledDecision.getDecisionRequirementsId())
+        .hasDecisionOutput("null")
```

Comment:
❓ Perhaps I misunderstand it, but should this be the string value `"null"`? I thought this should actually be `null`. (Sorry for mentioning this topic twice, but since I encountered it here again, I wanted to make sure we address it).

Reply:
Same as above. 

The value is a string representation of a JSON value. So, it is actually `null`.